src/components/PrintModal/WatermarkModal/WatermarkModal.js

```diff
@@ -0,0 +1,434 @@
+import ColorPalette from 'components/ColorPalette';
+import Slider from 'components/Slider';
+import { circleRadius } from 'constants/slider';
+import core from 'core';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { withTranslation } from 'react-i18next';
+import './WatermarkModal.scss';
+
+import ActionButton from 'components/ActionButton';
+
+const DESIRED_WIDTH = 300;
+const DESIRED_HEIGHT = 300;
+
+// numbers were taken from font dropdown menu in google docs
+const FONT_SIZES = [
+  8, 9, 10, 11, 12, 14, 16, 18, 24, 30, 36, 48,
+];
+
+const WATERMARK_LOCATIONS = {
+  CENTER: 'Center',
+  TOP_LEFT: 'Top Left',
+  TOP_RIGHT: 'Top Right',
+  TOP_CENTER: 'Top Center',
+  BOT_LEFT: 'Bottom Left',
+  BOT_RIGHT: 'Bottom Right',
+  BOT_CENTER: 'Bottom Center',
+};
+
+const FORM_FIELD_KEYS = {
+  location: 'location',
+  fontSize: 'fontSize',
+  text: 'text',
+  color: 'color',
+  opacity: 'opacity',
+};
+
+const DEFAULT_VALS = {
+  [FORM_FIELD_KEYS.location]: WATERMARK_LOCATIONS.CENTER,
+  [FORM_FIELD_KEYS.fontSize]: FONT_SIZES[FONT_SIZES.length / 2],
+  [FORM_FIELD_KEYS.text]: '',
+  [FORM_FIELD_KEYS.color]: new window.Annotations.Color(241, 160, 153),
+  [FORM_FIELD_KEYS.opacity]: 100,
+};
+
+class WatermarkModal extends React.PureComponent {
+  handleWatermarkRenderFxn;
+
+  static propTypes = {
+    isVisible: PropTypes.bool,
+    pageIndexToView: PropTypes.number,
+    modalClosed: PropTypes.func,
+    formSubmitted: PropTypes.func,
+    t: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    const locationSettings = this.initializeLocationSettings();
+    this.state = {
+      isVisible: false,
+      isColorPaletteVisible: false,
+      // eslint-disable-next-line object-shorthand
+      locationSettings: locationSettings,
+      previousLocationSettings: locationSettings,
```

Comment:
I'm not very clear about why we need this state. Is this related to the `watermarkModalOption` state in the PrintModal?

If yes can we use `watermarkModalOption` instead?

Reply:
when we press "Ok". those form field settings need to be saved.
so that next time we open up the watermark modal, we can retrieve those saved settings.

We need it for this other case:
1. user fools around with the form field settings but does not press ok and close the dialog.
2. user re-opens watermark modal. it should be able to display the previous saved setting.

let me know if that is unclear.